spanner/google/cloud/spanner/streamed.py

```diff
@@ -130,7 +159,35 @@ def consume_next(self):
 
         Parse the result set into new/existing rows in :attr:`_rows`
         """
-        response = six.next(self._response_iterator)
+        sleep_generator = retry.exponential_sleep_generator(
+            initial=retry._DEFAULT_INITIAL_DELAY,
+            maximum=retry._DEFAULT_MAXIMUM_DELAY)
```

Comment:
@jonparrott Should these names be public? Can we somehow avoid accessing the non-public surface in some other way?

Reply:
Not without another release-the-world danceathon, AFAIK.

Reply:
With #3891 this won't be necessary as we can just use a `Retry` object.

Reply:
@jonparrott I don't think we are going to want to do another release-the-world to get this spanner fix pushed out.

Reply:
@lukesneeringer Can you comment on whether we have to release the world to get a new version of `google-cloud-core` out (one which adds a forward-compatible feature to `retry.Retry`?)

Reply:
What do you mean "forward-compatible" in this case? If it is a non-breaking change to core, then we could just release 0.27.1 and no release-the-world would be necessary. If it is a breaking change, then yeah, we need to release the world.

Reply:
Dumb question, why not just send _nothing_? `initial` and `maximum` both have sensible defaults in `google.api.core` (those values).

Reply:
#3891 adds an optional `on_error` argument to `google.api.core.retry.retry_target` (and the `Retry` wrapper's `__call__`).  It is a semver break to call it `0.27.1` (a new feature, which means that code which uses it cannot revert to `0.27.0` if the change introduces a bug), but could be the pragmatic way to get the change deployed.

Reply:
@jonparrott 555bffa updates `consume_next` to use the `on_error` handling added in `core-0.27.1`.  PTAL